### PR TITLE
psm: code cleanup

### PR DESCRIPTION
**Pull request by gadfort**
No functional changes, just cleaning up the code in IRSolver to use fewer strings and instead pass in odb objects and make function names consistent with other tools.
Updates a few tests to account for long standing "diff" in ok files.

### CODE REVIEW

File: src/psm/src/pdnsim.tcl [hunk 1]
```diff
@@ -87,49 +81,50 @@ proc analyze_power_grid { args } {
...
 }
 
 sta::define_cmd_args "check_power_grid" {
+  [-error_file error_file]
   [-net power_net]}
 
 proc check_power_grid { args } {
   sta::parse_key_args "check_power_grid" args \
-    keys {-net} flags {}
+    keys {-net -error_file} flags {}
```
Comment (maliberty) [inline src/psm/src/pdnsim.tcl:113]: README update

**Comment by maliberty** [comment]
Please look at clang-tidy, especially bugprone-unchecked-optional-access

**Comment by maliberty** [comment]
Thanks for the big cleanup!

FYI @ahmadelrouby will start looking at the performance issues with matrix setup

**Comment by gadfort** [comment]
@maliberty thats great to hear. I have one more thing after this to address the handling of macro pins, but that should not be a big thing.

**Review by maliberty — APPROVED** [review]

**Pull request merged**